src/cli/cli.cpp

```diff
@@ -1491,10 +1491,9 @@ void Interpreter::HandleDnsAddressResponse(otError aError, const otDnsAddressRes
             OutputFormat(" TTL:%u ", ttl);
             index++;
         }
-
-        OutputLine("");
     }
 
+    OutputLine("");
```

Comment:
Will this impact tests and how they expect the output?

 I think the original DNS CLI output included error on same line as `"DNS response for {name} - "`

Reply:
> I think the original DNS CLI output included error on same line as `"DNS response for {name} - "`

Yes, which is actually incorrect.

This PR fixes this minor issue.